#**Trabajo final:**

**Valentina Velosa Garcia.**

In [3]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import statsmodels.api as sm

In [4]:
df = pd.read_csv("recdemand.csv")

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 659 entries, 0 to 658
Data columns (total 8 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   trips    659 non-null    int64  
 1   quality  659 non-null    int64  
 2   ski      659 non-null    object 
 3   income   659 non-null    int64  
 4   userfee  659 non-null    object 
 5   costC    659 non-null    float64
 6   costS    659 non-null    float64
 7   costH    659 non-null    float64
dtypes: float64(3), int64(3), object(2)
memory usage: 41.3+ KB


###**El diccionario de la base de datos implementado para el modelo es:**

trips : número de viajes en bote realizados en el lago Somerville

quality : percepción subjetiva de la calidad de 1 a 5

ski factor: pregunta acerca de si el individuo hizo o no water-ski en el lago Somerville

income: ingreso anual en miles de dólares 

userfee factor: pago de suscripción anual al lago 

costC: costo de visitar Lago Conroe (alternativa al lago Somerville) 

costS: costo de visitar Lago Somerville 

costH: costo de visitar Lago Houston (alternativa al lago Somerville) 

In [6]:
df = df[["trips", "quality","ski","income","userfee","costC", "costS", "costH"]]

In [7]:
df

trips  quality  ski  income userfee  costC   costS   costH
0        0        0  yes       4      no  67.59  68.620  76.800
1        0        0   no       9      no  68.86  70.936  84.780
2        0        0  yes       5      no  58.12  59.465  72.110
3        0        0   no       2      no  15.79  13.750  23.680
4        0        0  yes       3      no  24.02  34.033  34.547
..     ...      ...  ...     ...     ...    ...     ...     ...
654     40        5  yes       5     yes  25.63  18.916  29.684
655     40        3  yes       3      no  18.46  13.330  25.800
656     40        2  yes       4      no  58.84  47.000  62.760
657     50        4  yes       4      no  35.67  25.694  37.269
658     88        2   no       2      no  22.49   6.210  25.460

[659 rows x 8 columns]

In [8]:
df = df.replace({"ski" :{"yes" :1, "no":0 }})
df = df.replace({"userfee" :{"yes" :1, "no":0 }})

In [9]:
df

trips  quality  ski  income  userfee  costC   costS   costH
0        0        0    1       4        0  67.59  68.620  76.800
1        0        0    0       9        0  68.86  70.936  84.780
2        0        0    1       5        0  58.12  59.465  72.110
3        0        0    0       2        0  15.79  13.750  23.680
4        0        0    1       3        0  24.02  34.033  34.547
..     ...      ...  ...     ...      ...    ...     ...     ...
654     40        5    1       5        1  25.63  18.916  29.684
655     40        3    1       3        0  18.46  13.330  25.800
656     40        2    1       4        0  58.84  47.000  62.760
657     50        4    1       4        0  35.67  25.694  37.269
658     88        2    0       2        0  22.49   6.210  25.460

[659 rows x 8 columns]

#**Modelamiento Regresión Lineal**

In [10]:
y = df["trips"]
x = df[["quality","ski","income","userfee","costC", "costS", "costH"]]
x = sm.add_constant(x)
model = sm.OLS(y,x)
reg = model.fit()
reg.summary()

/usr/local/lib/python3.7/dist-packages/statsmodels/tsa/tsatools.py:117: FutureWarning: In a future version of pandas all arguments of concat except for the argument 'objs' will be keyword-only
  x = pd.concat(x[::order], 1)


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                  trips   R-squared:                       0.279
Model:                            OLS   Adj. R-squared:                  0.271
Method:                 Least Squares   F-statistic:                     36.02
Date:                Fri, 22 Apr 2022   Prob (F-statistic):           1.29e-42
Time:                        17:57:06   Log-Likelihood:                -2038.8
No. Observations:                 659   AIC:                             4094.
Df Residuals:                     651   BIC:                             4130.
Df Model:                           7                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          2.5615      0.572      4.479      0.000       1.439       3.685
quality        0.9134      0.125      7.305      0.000       0.668       1.159
ski            0.9992      0.462      2.161      0.031       0.091       1.907
income        -0.2071      0.119     -1.734      0.083      -0.442       0.027
userfee        8.2808      1.543      5.368      0.000       5.251      11.310
costC          0.0632      0.034      1.841      0.066      -0.004       0.131
costS         -0.1645      0.023     -7.236      0.000      -0.209      -0.120
costH          0.0895      0.028      3.177      0.002       0.034       0.145
==============================================================================
Omnibus:                      925.235   Durbin-Watson:                   0.378
Prob(Omnibus):                  0.000   Jarque-Bera (JB):           238996.003
Skew:                           7.395   Prob(JB):                         0.00
Kurtosis:                      95.115   Cond. No.                         938.
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

###**Podemos observar que las variables significativas con un P-Valor menor a 0.05 y que en el intervalo de confianza no contienen al cero, son:**
1. quality: que es la percepción subjetiva de la calidad de 1 a 5, con un P-Valor de 0.000
2. ski: que es la pregunta acerca de si hizo o no water-ski en el lago Somerville, con un P-Valor de 0.031  
3. userfee: que es el pago de subcripción anual al lago, con un P-Valor de 0.000
4. costS: Costo de visitar Lago Somerville, con un P-Valor de 0.000 
5. costH: Costo de visitar Lago Houston (alternativa), con un P-Valor de 0.002 

Cabe resaltar que un P-Valor menor al 5% me indica que se rechaza la Hipótesis Nula, en estos casos.

###**Las variables no significativas, es decir que tienen un P-Valor superior al 0.05 son:**

1. Income: que se refiere al ingreso anual en miles de dólares, con un P-Valor de 0.083
3. costC: que se refiere a la alternativa de viajar al lago Somerville, con un P-Valor de 0.066

Cabe resaltar que un P-Valor mayor al 5% me indica que se acepta la Hipótesis Nula, en estos casos.

###**Por otro lado, podemos analizar los coeficientes de las variables y cómo afectan a la variable discreta de estudio (Trips) en el modelo planteado.**

1. quality, incrementa en promedio 0.91 el número de viajes al lago. 
2. ski, incrementa en promedio 0.99 el número de viajes al lago. 
3. income, disminuye en promedio 0.20 el número de viajes al lago. 
4. userfee, incrementa en promedio 8.28 el número de viajes al lago. 
5. costC, incrementa apenas en promedio 0.06 el número de viajes al lago. 
6. costS, disminuye en promedio 0.16 el número de viajes al lago. 
7. costH, incrementa en promedio 0.08 el número de viajes al lago. 
